packages/plugin-scripts/src/manifest/descriptors.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2023 System Automation Corporation.
+// This file is licensed under the MIT License.
+
+export type WidgetDescriptor = {
+    id: string;
+    name: string;
+    description?: string;
+    version?: string;
+    src: string;
+    properties: WidgetPropertyDescriptor[];
+};
+
+export type WidgetPropertyDescriptor = {
+    name: string;
+    displayName: string;
+    type: string;
+    optional?: boolean;
```

Comment:
the property-name is **isOptional**

Reply:
@actong Yeah. The **isOptional** was added much before the **initialValue** (we can revisit this now). One of the things we need to look out for / think through is that: if a property is optional, it needs an initialValue (at-least with the way the code is structured right now in builder).
 Reasoning behind it:: The property values are usually set one level deep (on clicking the Edit icon). But if the setting is optional, it adds the property directly in the page configuration as soon as the toggle is switched ON(Mustard can then go to edit the value if desired). 
Alternative was that we add the property with 'undefined' , but that was causing complexities with when the toggle was switched off, and the property was removed from the page (which also corresponds to undefined value). So when Mustard would come back to the widget setting, it wasn't clear whether to keep the toggle ON or OFF. So in a nutshell, we decided to make sure there is a value for the setting every time the toggle was switched ON.

Top level **initialValue** was added much later when we identified a scenario where a default value was needed for a required setting as well.

Reply:
I tested with an optional text setting in a custom widget. I also configured an "initialValue" at the top level. At this point:
- the toggle seems to work the way we'd want to (there are minor glitches, but they're more about inconsistently treating an empty string as undefined or not. --- and its not noticeable/causing much issues). 
- the top level "initialValue" was being ignored -- I'll raise a PR in builder to handle that.

But it seems this PR doesn't need to be held up for that, as these changes would work as expected.

One minor thing though -- Please rename **optional** to **isOptional** to match what we've used everywhere so far.